vendor/diff-parsers-upstream-std.patch

```diff
@@ -0,0 +1,821 @@
++    /*empty*/                     { ([],Pcstr_tuple [],None) }
++  | OF constructor_arguments      { ([],$2,None) }
+   | COLON constructor_arguments MINUSGREATER atomic_type %prec below_HASH
+-                                  { ($2,Some $4) }
++                                  { ([],$2,Some $4) }
++  | COLON typevar_list DOT constructor_arguments MINUSGREATER atomic_type
++     %prec below_HASH
++                                  { ($2,$4,Some $6) }
+   | COLON atomic_type %prec below_HASH
+-                                  { (Pcstr_tuple [],Some $2) }
++                                  { ([],Pcstr_tuple [],Some $2) }
++  | COLON typevar_list DOT atomic_type %prec below_HASH
++                                  { ($2,Pcstr_tuple [],Some $4) }
+ ;
+ 
+ constructor_arguments:
+   | tys = inline_separated_nonempty_llist(STAR, atomic_type)
+     %prec below_HASH
+@@ -3184,12 +3195,12 @@
+       { $1 }
+ ;
+ %inline extension_constructor_declaration(opening):
+   d = generic_constructor_declaration(opening)
+     {
+-      let cid, args, res, attrs, loc, info = d in
+-      Te.decl cid ~args ?res ~attrs ~loc ~info
++      let cid, vars, args, res, attrs, loc, info = d in
++      Te.decl cid ~vars ~args ?res ~attrs ~loc ~info
+     }
+ ;
+ extension_constructor_rebind(opening):
+   opening
+   cid = mkrhs(constr_ident)
+--- ocaml-4.13-upstream/parsetree.mli
++++ ocaml-4.13/parsetree.mli
+@@ -140,10 +140,12 @@
+            - As the core_type of a Pexp_poly node.
+ 
+            - As the pld_type field of a label_declaration.
+ 
+            - As a core_type of a Ptyp_object node.
++
++           - As the pval_type field of a value_description.
+          *)
+ 
+   | Ptyp_package of package_type
+         (* (module S) *)
+   | Ptyp_extension of extension
+@@ -377,18 +379,20 @@
+            (module ME : S) is represented as
+            Pexp_constraint(Pexp_pack, Ptyp_package S) *)
+   | Pexp_open of open_declaration * expression
+         (* M.(E)
+            let open M in E
+-           let! open M in E *)
++           let open! M in E *)
+   | Pexp_letop of letop
+         (* let* P = E in E
+            let* P = E and* P = E in E *)
+   | Pexp_extension of extension
+         (* [%id] *)
+   | Pexp_unreachable
+         (* . *)
++  | Pexp_hole
++        (* _ *)
+ 
+ and case =   (* (P -> E) or (P when E0 -> E) *)
+     {
+      pc_lhs: pattern;
+      pc_guard: expression option;
+@@ -475,10 +479,11 @@
+ *)
+ 
+ and constructor_declaration =
+     {
+      pcd_name: string loc;
++     pcd_vars: string loc list;
+      pcd_args: constructor_arguments;
+      pcd_res: core_type option;
+      pcd_loc: Location.t;
+      pcd_attributes: attributes; (* C of ... [@id1] [@id2] *)
+     }
+@@ -524,15 +529,16 @@
+     ptyexn_loc: Location.t;
+     ptyexn_attributes: attributes; (* ... [@@id1] [@@id2] *)
+   }
+ 
+ and extension_constructor_kind =
+-    Pext_decl of constructor_arguments * core_type option
```

Comment:
What's this diff ?

Reply:
Also due to https://github.com/ocaml-ppx/ocamlformat/pull/1757